src/plugins/vis_augmenter/public/vega/helpers.ts

```diff
@@ -195,28 +205,34 @@ export const addPointInTimeEventsLayersToTable = (
   const augmentedTable = addMissingRowsToTableBounds(datatable, dimensions);
   const xAxisId = getXAxisId(dimensions, augmentedTable.columns);
 
-  if (isEmpty(visLayers)) return augmentedTable;
+  if (isEmpty(visLayers) || augmentedTable.rows.length === 0) return augmentedTable;
 
-  visLayers.every((visLayer: PointInTimeEventsVisLayer) => {
-    const visLayerColumnId = `${visLayer.pluginResource.id}`;
-    const visLayerColumnName = `${visLayer.pluginResource.name}`;
+  // Create columns for every unique event type. This is so we can aggregate on the different event types
+  // (e.g., 'Anomalies', 'Alerts')
+  [
+    ...new Set(visLayers.map((visLayer: PointInTimeEventsVisLayer) => visLayer.pluginEventType)),
```

Comment:
Why the `Set` here? Is the idea that multiple plugins may contribute vis layer data of the same pluginEventType?

Reply:
That, or one plugin contributes vis layer data of a different pluginEventType. So just deduping in case of any of those scenarios